engine/access/rest/websockets/data_providers/account_statuses_provider.go

```diff
@@ -169,42 +167,28 @@ func parseAccountStatusesArguments(
 	args.StartBlockID = startBlockID
 	args.StartBlockHeight = startBlockHeight
 
-	// Parse 'event_types' as a JSON array
-	var eventTypes parser.EventTypes
-	if eventTypesIn, ok := arguments["event_types"]; ok && eventTypesIn != "" {
-		result, err := common.ParseInterfaceToStrings(eventTypesIn)
-		if err != nil {
-			return accountStatusesArguments{}, fmt.Errorf("'event_types' must be an array of string")
-		}
-
-		err = eventTypes.Parse(result)
-		if err != nil {
-			return accountStatusesArguments{}, fmt.Errorf("invalid 'event_types': %w", err)
-		}
+	// Parse 'heartbeat_interval' argument
+	heartbeatInterval, err := extractHeartbeatInterval(arguments)
+	if err != nil {
+		return accountStatusesArguments{}, err
 	}
+	args.HeartbeatInterval = heartbeatInterval
 
-	// Parse 'accountAddresses' as []string{}
-	var accountAddresses []string
-	if accountAddressesIn, ok := arguments["account_addresses"]; ok && accountAddressesIn != "" {
-		accountAddresses, err = common.ParseInterfaceToStrings(accountAddressesIn)
-		if err != nil {
-			return accountStatusesArguments{}, fmt.Errorf("'account_addresses' must be an array of string")
-		}
+	// Parse 'event_types' as a JSON array
+	eventTypeList, err := extractArrayOfStrings(arguments, "event_types", false)
+	if err != nil {
+		return accountStatusesArguments{}, err
 	}
 
-	var heartbeatInterval uint64
-	if heartbeatIntervalIn, ok := arguments["heartbeat_interval"]; ok && heartbeatIntervalIn != "" {
-		result, ok := heartbeatIntervalIn.(string)
-		if !ok {
-			return accountStatusesArguments{}, fmt.Errorf("'heartbeat_interval' must be a string")
-		}
-
-		heartbeatInterval, err = util.ToUint64(result)
-		if err != nil {
-			return accountStatusesArguments{}, fmt.Errorf("invalid 'heartbeat_interval': %w", err)
-		}
+	eventTypes, err := parser.NewEventTypes(eventTypeList)
```

Comment:
this validation logic is already done within `NewAccountStatusFilter`, so you don't need to do it initially here. a list of strings is fine